src/library/exporttrackmetadatainfo.cpp

```diff
@@ -5,12 +5,18 @@
 
 namespace mixxx {
 
+//static
+bool ExportTrackMetadataInfo::s_bShownOncePerSession = false;
+
 void ExportTrackMetadataInfo::showMessageBox() {
-    QMessageBox::information(
-            nullptr,
-            tr("Export Modified Track Metadata"),
-            tr("Mixxx may wait to modify files until it is certain that writing into those files will not cause audible glitches. "
-                    "If you do not see changed metadata in other programs, close Mixxx to modify the files immediately."));
+    if (!s_bShownOncePerSession) {
+        QMessageBox::information(
+                nullptr,
+                tr("Export Modified Track Metadata"),
+                tr("Mixxx may wait to modify files until it is certain that writing into those files will not cause audible glitches. "
+                        "If you do not see changed metadata in other programs, close Mixxx to modify the files immediately."));
```

Comment:
Now that the TrackDAO cache has been removed, do we really need this notice anymore?

Reply:
Or should we modify it to say "Mixxx may wait to modify files until they are not loaded to any decks in Mixxx. If you do not see changed metadata in other programs immediately, eject the track from all decks in Mixxx or shutdown Mixxx."

Reply:
I've use your text (edit: "Mixxx" only once per sentence) and did some internal renamings to let this workaround look less ugly.

The user experience should be much better now! It was a lot of work, but I hope it is worth it :) I'm glad that we could even get rid of this obscure cache in TrackDAO. It helps to keep the library synchronized more closely and saves shutdown time by executing database updates sooner.